File: IPWorks Samples/GET URL/javascript/httpurl.js
/*
 * IPWorks 2022 JavaScript Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworks
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
 
const readline = require("readline");
const ipworks = require("@nsoftware/ipworks");

if(!ipworks) {
  console.error("Cannot find ipworks.");
  process.exit(1);
}
let rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

main();

async function main() {
	const argv = process.argv;
	if(argv.length !== 3) {
		console.log("Usage: node httpurl.js url");
		console.log("  url the url to fetch");
		console.log("Example: node httpurl.js https://www.google.com");
		process.exit();
	}

	const url = argv[2];

	const http = new ipworks.http();


	http.on("Transfer", function(e){
		console.log(e.text.toString());	
	})
	.on("SSLServerAuthentication", function(e) {
		e.accept = true;
	});

	http.setFollowRedirects(1);

	await http.get(url).catch((err) => {
		console.log(err.message);
	});
	process.exit();
}


function prompt(promptName, label, punctuation, defaultVal)
{
  lastPrompt = promptName;
  lastDefault = defaultVal;
  process.stdout.write(`${label} [${defaultVal}] ${punctuation} `);
}

File: IPWorks Samples/GET URL/java/httpurl.java
/*
 * IPWorks 2022 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworks
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import ipworks.*;


public class httpurl {
	
	public static void main(String[] args) {
	  
	  if (args.length != 1) {

			System.out.println("usage: httpurl url");
			System.out.println("");
			System.out.println("  url  the url to fetch");
			System.out.println("\r\nExample: httpurl https://www.google.com");

		} else {
			
			Http http = new Http();					
			try {
				http.addHttpEventListener(new DefaultHttpEventListener(){
					
					public void SSLServerAuthentication(HttpSSLServerAuthenticationEvent arg0) {
						arg0.accept = true; //this will trust all certificates and it is not recommended for production use
					}
					public void transfer(HttpTransferEvent e) {
						System.out.print(new String(e.text));
					}
		        });
				http.setFollowRedirects(2);
			    http.get(args[0]); //url
			} catch (Exception ex) {
				System.out.println(ex.getMessage());
			}			
		}	  	  
	}
}

class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }

  static String prompt(String label, String punctuation, String defaultVal)
  {
	System.out.print(label + " [" + defaultVal + "] " + punctuation + " ");
	String response = input();
	if(response.equals(""))
		return defaultVal;
	else
		return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksException) {
      System.out.print(" (" + ((IPWorksException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }
}



